src/EditorFeatures/Test/Extensions/SourceTextContainerExtensionsTests.cs

```diff
@@ -0,0 +1,37 @@
+﻿using Microsoft.CodeAnalysis.Text;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.UnitTests.Extensions
+{
+    public class SourceTextContainerExtensionsTests
+    {
+        [Fact]
+        public void GetBufferTextFromNonTextContainerThrows()
+        {
+            var containerMock = new Mock<SourceTextContainer>();
+            Assert.Throws<ArgumentException>(containerMock.Object.GetTextBuffer);
```

Comment:
Ahh, both @rchande and I were tricked by the extension method, so switching to the more obvious invocation would probably prevent others from being confused as well.